cmd/replay/main.go

```diff
@@ -311,6 +312,11 @@ func main() {
 			}
 			espressocrypto.VerifyNamespace(chainConfig.ChainID.Uint64(), *jst.Proof, *jst.Header.PayloadCommitment, *jst.Header.NsTable, txs)
 			espressocrypto.VerifyMerkleProof(jst.BlockMerkleJustification.BlockMerkleProof.Proof, jsonHeader, *jst.BlockMerkleJustification.BlockMerkleComm, commitment)
+		} else if validatingAgainstCentrailized {
+			l1Block := message.Message.Header.BlockNumber
```

Comment:
I think that there is a potential censorship issue here. This block number is set at the discretion of the sequencer [here](https://github.com/EspressoSystems/nitro-espresso-integration/blob/integration/execution/gethexec/espresso_sequencer.go#L89). So I think we'll need to remember to add some logic to throw an error if the L1 block number passed into GetHotShotAvailability is too far in the future, otherwise the sequencer can set this to some arbitrarily high height and bypass espresso validation.

Reply:
Agree. We can rethink about this when we know more about how the L1 head affects the validation pipeline. Intuitively there should be some errors occur if the l1 height is incorrect.